src/scheduler/ProcessList.java

```diff
@@ -34,17 +34,21 @@ public static PCB removeNextInReadyList() {
 		}
 	}
 
-	public static void decreaseBlockedListWait(boolean roundRobin) {
+	public static void decreaseBlockedListWait() {
 		if (blockedList.size() > 0) {
 			for (int i = 0; i < blockedList.size(); i++) {
 				PCB pcb = blockedList.remove(i);
 				pcb.decreaseWait();
 				if (pcb.getWait() == 0) {
 					pcb.getProcess().setState(State.READY);
-					if (pcb.getCredit() > 0 || !roundRobin) {
+					if (pcb.getCredit() > 0) {
 						addReadyProcess(pcb);
 					} else {
-						addReadyProcessInLastPosition(pcb);
+						if (allProcessInReadyListWithZEROCredit()) {
```

Comment:
You are spending time calculating this boolean, when it does same thing as before. You should keep variable `roundRobin`. Also I think this logic doesn't make sense. You made a refact on if, but it does same thing as before. Better to keep previous if.